errors/CycleInDeclaration.md

```diff
@@ -39,3 +39,45 @@ Note that cycles can also spring up in much less obvious ways, e.g. if you defin
 ## Notes
 
 - If you are trying to implement a recursive parser in PureScript, then you might find this post on [recursive parsing in PureScript](https://github.com/Thimoteus/SandScript/wiki/2.-Parsing-recursively) to be helpful.
+
+### Additional Examples
+
+Mutually recursive functions:
```

Comment:
If we move the generic deriving example to be a link then we can change this header to be `### Mutual recursion` and just focus on this mutual recursion example. I think that fits better with how [other errors use the Notes section](https://github.com/purescript/documentation/blob/2059f1d272aa863c465bcd4b0ade815a0bdfccc4/errors/KindsDoNotUnify.md).

Also, what do you think of trimming the example a little bit to something like this?

```purs
isEven :: Int -> Boolean
isEven n 
  | n == 0 = true
  | otherwise = again (n - 1)

again :: Int -> Boolean
again = isEven
```

which also fails, producing this error:

```
  The value of isEven is undefined here, so this reference is not allowed.

while checking that expression isEven
  has type Int -> Boolean
in binding group isEven, again
```